doxy

```diff
@@ -3,122 +3,52 @@
 if [ -f ${doxyfile} ]
 	then
 		doxygen ${doxyfile}
 		retVal=$?
 		if [ $retVal -ne 0 ]
 		then
-            echo "Error: exit code ${retVal} on ${doxyfile}."
-            exit $retVal
-        fi
-fi
-
-mkdir /tmp/bill
-
-doxyfile="dox/books/reference_manual/DoxyRefman"
-if [ -f ${doxyfile} ]
-	then
-		doxygen ${doxyfile}
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo "Error: exit code ${retVal} on ${doxyfile}."
-            exit $retVal
-        fi
-		cd    /tmp/bill/refman/latex
-		make
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo "Error: exit code ${retVal} on Make of ${doxyfile}."
-            exit $retVal
-        fi
-		cd $here
-        mv /tmp/bill/refman/latex/refman.pdf docs/Bill_Reference_Manual.pdf
-fi
-
-doxyfile="dox/DoxyUserman"
-if [ -f ${doxyfile} ]
-	then
-		doxygen ${doxyfile}
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo "Error: exit code ${retVal} on ${doxyfile}."
+            echo "Error: exit code ${$retVal} on ${doxyfile}."
             exit $retVal
         fi
-# 		cd docs/userman/latex
-# 		make
-# 		retVal=$?
-# 		if [ $retVal -ne 0 ]
-# 		then
-#             echo "Error: exit code ${retVal} on Make of ${doxyfile}."
-#             exit $retVal
-#       fi
-#       mv refman.pdf ../../Bill_User_Manual.pdf
-# 		cd $here
 fi
 
-doxyfile="dox/books/Bill_Primer/DoxyLangman"
-if [ -f ${doxyfile} ]
-	then
-		doxygen ${doxyfile}
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo"Error: exit code ${retVal} on ${doxyfile}."
-            exit $retVal
-        fi
-		cd /tmp/bill/langman/latex
-		make
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo "Error: exit code ${retVal} on Make of ${doxyfile}."
-            exit $retVal
-        fi
-		cd $here
-        mv /tmp/bill/langman/latex/refman.pdf docs/Bill_Primer.pdf
-fi
-
-doxyfile="dox/DoxyStlman"
-if [ -f ${doxyfile} ]
-	then
-		doxygen ${doxyfile}
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo "Error: exit code ${retVal} on ${doxyfile}."
-            exit $retVal
-        fi
-# 		cd docs/stlman/latex
-# 		make
-# 		retVal=$?
-# 		if [ $retVal -ne 0 ]
-# 		then
-#             echo "Error: exit code ${retVal} on Make of ${doxyfile}."
-#             exit $retVal
-#       fi
-#       mv refman.pdf ../../Bill_Standard_Library_Reference.pdf
-# 		cd $here
-fi
-
-# rm -r /tmp/bill
-
-# echo ${here}
+# pdf docs
+for i in {0..3}
+do
+    doxyfile="dox/books/${books[i]}/${doxyfiles[i]}"
+    if [ -f ${doxyfile} ]
+        then
+            doxygen ${doxyfile}
+            retVal=$?
+            if [ $retVal -ne 0 ]
+            then
+                echo "Error: exit code ${$retVal} on doxygen ${doxyfile}."
+                exit $retVal
+            fi
+            cd /tmp/bill/${locations[i]}/latex
+            make
+            retVal=$?
+            if [ $retVal -ne 0 ]
+            then
+                echo "Error: exit code ${retVal} on Make of ${doxyfile}."
+                exit $retVal
```

Comment:
You need to fix directory location before you exit.

Reply:
fixed.